File: Web/Scripts/js/angular/controller/compania/companiaController.js
app.controller('companiaController', ['$rootScope', '$scope', 'companiaService', function ($rootScope, $scope, companiaService) {
    
    $scope.model = {};
    $scope.elementos = { lista: [] };

    $scope.desabilitarCampos = true;

    $scope.guardarCompania = function () {
        var params = {
            companiaViewModel: $scope.model
        };
        if ($scope.desabilitarCampos == false) {
            companiaService.guardarCompania(params).then(function (data) {
                if (data.data.result_description) {
                    $scope.messageSuccess(data.data.result_description);
                    $scope.desabilitarCampos = true;
                } else {
                    $scope.messageError(data.data.error);
                }
            });
        } else {
            $.notify({
                icon: 'fa fa-exclamation-circle',
                title: 'Alerta !',
                message: 'Tiene que estar activadas las casillas para poder guardar cambios.'
            }, { type: 'warning', z_index: 2000 });
        }
    }

    $scope.listarCompanias = function () {
        companiaService.listarCompania().then(function (data) {
            if (data.data) {
                $scope.elementos.lista = data.data;
                $scope.model.idCompania = $scope.elementos.lista[0].idCompania;
                $scope.model.razonSocial = data.data[0].razonSocial;
                $scope.model.ruc = data.data[0].ruc;
                $scope.model.domicilioFiscal = data.data[0].domicilioFiscal;
                $scope.model.correo = data.data[0].correo;

                $scope.model.idTitular = $scope.elementos.lista[0].idTitular;
                $scope.model.titularNombre = data.data[0].titularNombre;
                $scope.model.titularApellidos = data.data[0].titularApellidos;
                $scope.model.titularCorreo = data.data[0].titularCorreo;
                $scope.model.titularTelefono = data.data[0].titularTelefono;
                $scope.model.titularCargo = data.data[0].titularCargo;

                $scope.model.idContacto = $scope.elementos.lista[0].idContacto;
                $scope.model.contactoNombre = data.data[0].contactoNombre;
                $scope.model.contactoApellidos = data.data[0].contactoApellidos;
                $scope.model.contactoCorreo = data.data[0].contactoCorreo;
                $scope.model.contactoTelefono = data.data[0].contactoTelefono;
                $scope.model.contactoCargo = data.data[0].contactoCargo;
            } else {
                $scope.elementos.lista = data.data;
            }            
        });
    }
    
    $scope.editarCompania = function () {
        $scope.desabilitarCampos = false;
    }

    $scope.listarCompanias();

}]);

File: Web/Controllers/CompaniaController.cs
using System;
using System.Collections.Generic;
using System.Web.Mvc;
using Web.Models.Compania;
using Web.Utilitario;
using Negocio;
using Datos;

namespace Web.Controllers
{
    public class CompaniaController : BaseController
    {
        CompaniaNEG companiaNEG = new CompaniaNEG();
        PersonaMastNEG personaMastNEG = new PersonaMastNEG();

        // GET: Compania
        [Authorize(Roles = "ADMINISTRADOR")]
        #region Agregar Compañia
        public ActionResult AddCompania()
        {
            return View();
        }

        public JsonResult guardarCompania(RegistroCompaniaViewModel companiaViewModel)
        {
            long IdTitular = 0;
            long IdContacto = 0;

            try
            {
                OperationResult resultado = null;
                if (companiaViewModel.idCompania != 0)
                {
                    resultado = personaMastNEG.actualizarPersonaMast(companiaViewModel.idTitular,companiaViewModel.titularNombre, companiaViewModel.titularApellidos, 1, 2, null, companiaViewModel.titularCargo, companiaViewModel.titularCorreo, companiaViewModel.titularTelefono, null, "ACTIVO");
                    IdTitular = resultado.data;
                    resultado = personaMastNEG.actualizarPersonaMast(companiaViewModel.idContacto,companiaViewModel.contactoNombre, companiaViewModel.contactoApellidos, 1, 2, null, companiaViewModel.contactoCargo, companiaViewModel.contactoCorreo, companiaViewModel.contactoTelefono, null, "ACTIVADO");
                    IdContacto = resultado.data;
                    resultado = companiaNEG.actualizarCompania(companiaViewModel.idCompania,companiaViewModel.razonSocial, companiaViewModel.ruc, companiaViewModel.domicilioFiscal, companiaViewModel.correo, Convert.ToInt32(IdTitular), Convert.ToInt32(IdContacto), IdUsuario());
                }
                else
                {
                    resultado = personaMastNEG.guardarPersonaMast(companiaViewModel.titularNombre, companiaViewModel.titularApellidos, 1, 2, null, companiaViewModel.titularCargo, companiaViewModel.titularCorreo, companiaViewModel.titularTelefono,null, "ACTIVO");
                    IdTitular = resultado.data;
                    resultado = personaMastNEG.guardarPersonaMast(companiaViewModel.contactoNombre, companiaViewModel.contactoApellidos, 1, 2, null, companiaViewModel.contactoCargo, companiaViewModel.contactoCorreo, companiaViewModel.contactoTelefono,null, "ACTIVADO");
                    IdContacto = resultado.data;
                    resultado = companiaNEG.guardarCompania(companiaViewModel.razonSocial, companiaViewModel.ruc, companiaViewModel.domicilioFiscal, companiaViewModel.correo, Convert.ToInt32(IdTitular), Convert.ToInt32(IdContacto), IdUsuario());
                }

                Util.verificarError(resultado);
                return Json(new { code_result = resultado.code_result, data = resultado.data, result_description = resultado.title });
            }
            catch (Exception e)
            {
                HttpContext.Response.StatusCode = 500;
                return Json(Util.errorJson(e));
            }
        }
        #endregion

        [Authorize(Roles = "ADMINISTRADOR")]
        #region Listar Compañia
        public ActionResult ViewAllCompania()
        {
            return View();
        }

        public JsonResult listarCompania()
        {
            try
            {
                List<Compania> resultado = companiaNEG.listarCompania();

                List<MostrarCompaniaViewModel> lista = MostrarCompaniaViewModel.convert(resultado);
                return Json(lista);
            }
            catch (Exception e)
            {
                HttpContext.Response.StatusCode = 500;
                return Json(Util.errorJson(e));
            }
        }
        #endregion
               
    }
}